lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -2,11 +2,21 @@
 
 module Reports
   class SiegeReport
-    def initialize(building:)
-    end
+    def self.call(building:)
+      # this should stop the method if getting null
+      return unless building
```

Comment:
To sprawdzenie spokojnie by się mogło znaleźć gdzieś wyżej, żeby nie obciążać logiki serwisu :) Serwis najlepiej jeśli ma jedną odpowiedzialność, a Twój serwis poza liczeniem ile dni przetrwa dany budynek, stwierdza upadłość budynku jeśli nie ma w nim wojowników :)